front50-core/src/main/groovy/com/netflix/spinnaker/front50/model/pluginartifact/DefaultPluginArtifactRepository.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.front50.model.pluginartifact;
+
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.front50.model.ObjectKeyLoader;
+import com.netflix.spinnaker.front50.model.ObjectType;
+import com.netflix.spinnaker.front50.model.StorageService;
+import com.netflix.spinnaker.front50.model.StorageServiceSupport;
+import com.netflix.spinnaker.kork.exceptions.IntegrationException;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import rx.Scheduler;
+
+public class DefaultPluginArtifactRepository extends StorageServiceSupport<PluginArtifact>
+    implements PluginArtifactRepository {
+  public DefaultPluginArtifactRepository(
+      StorageService service,
+      Scheduler scheduler,
+      ObjectKeyLoader objectKeyLoader,
+      long refreshIntervalMs,
+      boolean shouldWarmCache,
+      Registry registry) {
+    super(
+        ObjectType.PLUGIN_ARTIFACT,
+        service,
+        scheduler,
+        objectKeyLoader,
+        refreshIntervalMs,
+        shouldWarmCache,
+        registry);
+  }
+
+  @Nonnull
+  @Override
+  public Collection<PluginArtifact> getByService(@Nonnull String service) {
+    return all().stream()
+        .filter(artifact -> hasActiveReleaseMatchingService(artifact, service))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public PluginArtifact create(String id, PluginArtifact item) {
+    Objects.requireNonNull(item.getId());
+    if (!item.getId().equals(id)) {
```

Comment:
very minor. Is the comment about Orca relevant here ..?

Reply:
Yes, it is. The Orca bits haven't been done yet, but they will need to be.